src/packet/signature/types.rs

```diff
@@ -822,11 +859,8 @@ pub enum SubpacketData {
     TrustSignature(u8, u8),
     RegularExpression(BString),
     ExportableCertification(bool),
-    IssuerFingerprint(
-        KeyVersion,
-        #[debug("{}", hex::encode(_1))] SmallVec<[u8; 20]>,
-    ),
-    PreferredAeadAlgorithms(SmallVec<[AeadAlgorithm; 2]>),
+    IssuerFingerprint(KeyVersion, #[debug("{}", hex::encode(_1))] Vec<u8>),
```

Comment:
why not a smallvec anymore?

Reply:
Good question. I changed Fingerprints to use Vec instead of SmallVec, because we now need Fingerprints to be dynamically either 20 or 32 bytes long, and I wasn't sure if that's reasonable to do with SmallVec.

If it is, I should change Fingerprints back to using smallvec.

Reply:
I wonder if we shouldn't just do sth like this

```rust
pub enum Fingerprint {
    V4([u8; 20]),
    V6([u8; 32]),
}
```

Reply:
> I wonder if we shouldn't just do sth like this
> 
> ```rust
> pub enum Fingerprint {
>     V4([u8; 20]),
>     V6([u8; 32]),
> }
> ```

I've taken a stab at such a change in 4e6dc901ba583406fcb3c1683219d2a827895ada, and would love to hear your thoughts on the direction.

The change feels in part reasonable, to me. But in some places we now have redundancy between the fingerprint type (which encodes a key version), and a separate key version field. I haven't yet decided how I feel about that.

Reply:
overall I think this makes the state strictly better